src/main/kotlin/nikolay/egorov/cli/analysis/lexical/Lexer.kt

```diff
@@ -0,0 +1,101 @@
+package nikolay.egorov.cli.analysis.lexical
+
+import nikolay.egorov.cli.analysis.exceptions.LexException
+
+/**
+ * Base class for lexical analysis phase
+ * @see LexerInterface
+ */
+class Lexer(givenString: String) : LexerInterface {
```

Comment:
```suggestion
class Lexer(val givenString: String) : LexerInterface {
```

Тогда fedString не нужна